src/node-file-trace.ts

```diff
@@ -152,8 +158,10 @@ export class Job {
   async readlink (path: string) {
     const cached = this.symlinkCache.get(path);
     if (cached !== undefined) return cached;
+    await this.fileIOQueue.acquire();
     try {
       const link = await fsReadlink(path);
+      this.fileIOQueue.release();
```

Comment:
I think we should move `release()` out of `try` and `catch` and instead move it into `finally` because I think there is a scenario where `release()` could be called twice and cause bugs.

Reply:
There is `return` in the try block, so we can't put the release in the `finally`. There is only the file op in the try block that may fail, so I think it would be okay that put release both in the try and catch block.

Reply:
TIL